controllers/users_test.go

```diff
@@ -46,40 +52,157 @@ func createUserTestData() (*test.OrderedTests, error) {
 		AssetsID:   assetID,
 	}
 
-	testCase := "no_existing_user"
-	data := test.Data{
-		Data:     make(map[string]interface{}),
-		Expected: make(map[string]interface{}),
+	nomineeID, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
 	}
 
-	data.Data.(map[string]interface{})["input"] = user
-	data.Data.(map[string]interface{})["db_mock"] = nil
-	data.Expected.(map[string]interface{})["data"] = &user
-	data.Expected.(map[string]interface{})["error"] = nil
-	dataSet.TestDataSet[testCase] = data
-	dataSet.OrderedList = append(dataSet.OrderedList, testCase)
-
-	testCase = "existing_user"
-	data = test.Data{
-		Data:     make(map[string]interface{}),
-		Expected: make(map[string]interface{}),
+	productID, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
+	}
+
+	usersProducts := models.UserProducts{
+		ProductMap: make(map[uuid.UUID]int),
 	}
 
-	data.Data.(map[string]interface{})["input"] = user
-	data.Data.(map[string]interface{})["db_mock"] = &user
-	data.Expected.(map[string]interface{})["data"] = nil
-	data.Expected.(map[string]interface{})["error"] = ErrDuplicateEmailEntry
-	dataSet.TestDataSet[testCase] = data
-	dataSet.OrderedList = append(dataSet.OrderedList, testCase)
+	product := models.Product{
+		ID:     productID,
+		Public: true,
+	}
+
+	privileges := make(models.Privileges, 2)
+	privileges[0].ID = 0
+	privileges[0].Name = "Owner"
+	privileges[0].Description = "description0"
+	privileges[1].ID = 1
+	privileges[1].Name = "User"
+	privileges[1].Description = "description1"
+
+	switch testID {
+	case CreateUserTest:
+		testCase := "no_existing_user"
+		data := test.Data{
+			Data:     make(map[string]interface{}),
+			Expected: make(map[string]interface{}),
+		}
+
+		data.Data.(map[string]interface{})["input"] = user
```

Comment:
You could/should leave the type cast and use the following initialization instead:
```suggestion
                data         := make(map[string]interface{})
                expected  := make(map[string]interface{})


		data["input"] = user
		expected["data"] = &user
		...
		dataSet.TestDataSet[testCase] = test.Data{
			Data:         data,
			Expected:  expected,
		}
```